### PR TITLE
Bumping to v3.0 - PaginatedResponseV2

### PR DESCRIPTION
**Pull request by abellotti**
Bumping up Topological Inventory Api to v3.0
- Support for PaginatedResponseV2 for both REST and GraphQL
- Adding v3 specs

Fixes: TPINVTRY-864

### CODE REVIEW

**Comment by lpichler** [comment]
@abellotti 

- when this will be merged, I guess we don't need 2.1 https://github.com/RedHatInsights/topological_inventory-api/pull/285 ?

File: app/controllers/api/v3x0.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module Api
+  module V3x0
+    class RootController < ApplicationController
+      def openapi
```
Comment (lpichler) [inline app/controllers/api/v3x0.rb]: @abellotti 

Is here reason to not leverage controllers from previous version like it is here 
https://github.com/lpichler/topological_inventory-api/blob/b1078f434b15e7db041721fd17805f85150d22e0/app/controllers/api/v2x0.rb#L10
?
and to have in v3.0 controllers  just `Mixins::IndexMixin` ?

thanks!

Comment (bdunne) [reply app/controllers/api/v3x0.rb]: @lpichler We need to pull in the `Api::V3x0::Mixins::IndexMixin`, not the one from `V1x0`

File: app/controllers/api/v3x0/mixins/index_mixin.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Api
+  module V3x0
+    module Mixins
+      module IndexMixin
+        def index
+          raise_unless_primary_instance_exists
+          render :json => Insights::API::Common::PaginatedResponseV2.new(
+            :base_query => scoped(filtered.where(params_for_list)),
```
Comment (bdunne) [inline app/controllers/api/v3x0/mixins/index_mixin.rb]: @abellotti Does this need to inherit from the `V1x0` mixin for the `scoped` method?

Comment (bdunne) [reply app/controllers/api/v3x0/mixins/index_mixin.rb]: Nevermind, I see, the controller inheritance is taking care of it.

File: app/controllers/api/v3x0.rb (hunk 1 above)
Comment (lpichler) [reply app/controllers/api/v3x0.rb]: 👍 👍

**Pull request merged**